src/mmrelay/config.py

```
@@ -71,6 +71,60 @@ def __init__(self) -> None:
+    """Emit deprecation warnings for legacy credentials_path / store_path overrides.
+
+    Warns once per process when any of the following are present:
+    - top-level ``credentials_path`` in config
+    - ``matrix.credentials_path`` in config
+    - ``matrix.e2ee.store_path`` or ``matrix.encryption.store_path`` in config
+    - ``MMRELAY_CREDENTIALS_PATH`` environment variable
+
+    These overrides are scheduled for removal; ``MMRELAY_HOME`` is the
+    supported way to control data paths.
+    """
+    global _legacy_path_override_warning_shown
```

Ruff (0.15.10)

[warning] Using the global statement to update `_legacy_path_override_warning_shown` is discouraged

(PLW0603)